### PR TITLE
ref(js): Drop types/index.tsx barrel file

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

**Comment by priscilawebdev** [comment]
why are we doing this? Is it something we no longer need?

File: biome.json [hunk 1]
```diff
@@ -40,9 +40,7 @@
         "noRestrictedImports": {
           "level": "warn",
           "options": {
-            "paths": {
-              "sentry/types": "Please import directly. For example: import type {Organization} from 'sentry/types/organization'"
-            }
+            "paths": {}
           }
         }
```
Comment (scttcper) [inline biome.json:45]: i think its no longer used

**Review by scttcper — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
> why are we doing this? Is it something we no longer need?

A while back @anonrig found that avoiding this barrel file improved typescript performance (?) and started moving everything to imports from their sepcific type files. I finished that in #76150 so we can delete this file

File: biome.json (hunk 1 above)
Comment (evanpurkhiser) [reply biome.json:45]: was just gonna leave it if we wanted to easily add more, idk

**Pull request merged**